Source/cmMakefileTargetGenerator.cxx

```diff
@@ -1748,11 +1748,50 @@ void cmMakefileTargetGenerator::AppendObjectDepends(
                                          this->BuildFileNameFull.c_str());
 }
 
+void cmMakefileTargetGenerator::AppendLinkDependsForJson(
+        std::vector<std::string>& depends, const std::string& linkLanguage)
+{
+    // Add dependencies on the compiled object files.
+    std::string const& fullPath =
```

Comment:
Specify the meaning of what is the path in the name. Is it a path to project build directory or something else? In this case, name it buildDirFullPath